client/components/jetpack/new-year-2021-sale-banner/index.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { useTranslate } from 'i18n-calypso';
+import { Moment } from 'moment';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+
+/**
+ * Style dependencies
+ */
+import desktopBanner from './new-year-banner.png';
+import desktopBannerRetina from './new-year-banner-2x.png';
+import mobileBanner from './new-year-banner_mobile.png';
+import mobileBannerRetina from './new-year-banner_mobile-2x.png';
+
+const Banner: React.FC = () => {
+	const translate = useTranslate();
+
+	return (
+		<div className="new-year-2021-sale-banner">
+			<picture className="new-year-2021-sale-banner__image">
+				<source
+					media="(max-width: 679px)"
+					srcSet={ `${ mobileBanner } 375w, ${ mobileBannerRetina } 750w` }
+				/>
+				<source
+					media="(min-width: 680px)"
+					srcSet={ `${ desktopBanner } 1080w, ${ desktopBannerRetina } 2160w` }
+				/>
+				<img
+					alt={ translate(
+						'New Year 2021 sale! Save %(percent)d%% at checkout with code %(code)s through January 18',
+						{
+							args: { percent: 40, code: 'NEWPACK' },
+						}
+					) }
+					src={ desktopBanner }
+				/>
+			</picture>
+		</div>
+	);
+};
+
+// Something of an experiment here; maybe this is something we can adapt for future promos?
+const TimeWindowWrapper: React.FC< TimeWindowProps > = ( { start, end, children } ) => {
+	const moment = useLocalizedMoment();
+
+	// '[)': The checked range includes the 'start' value,
+	// but ends just before the 'end' value
+	const showPromo = moment().isBetween( start, end, 'second', '[)' );
+
+	return showPromo ? <>{ children }</> : null;
+};
+
+type TimeWindowProps = {
+	start: Moment;
+	end: Moment;
+};
```

Comment:
Considering that we'll probably run more of these time-boxed promotions, I wonder if it would be good to have a common, testable wrapper that can handle the date math for us?

Reply:
I think that is a great idea.